### PR TITLE
feat(cmd/influxd): export 1.x continuous queries as part of upgrade

### PR DESCRIPTION
**Pull request by danxmoran**
Part of #19985 

Write CQs to disk as databases are upgraded. The issue didn't describe an encoding scheme for the exported queries. I went with TSV as I expected that tab wouldn't clash with the content of the queries themselves, but I'm not sure if it's the most useful option.

- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Feature flagged (if modified API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: cmd/influxd/upgrade/database.go [hunk 1]
```diff
@@ -174,6 +187,16 @@ func upgradeDatabases(ctx context.Context, v1 *influxDBv1, v2 *influxDBv2, v1opt
 				log.Warn("Empty retention policy")
 			}
 		}
+
+		for _, cq := range db.ContinuousQueries {
+			if options.verbose {
+				log.Info("Exporting CQ", zap.String("db", db.Name), zap.String("cq_name", cq.Name))
+			}
+			_, err := cqFile.WriteString(fmt.Sprintf("%s\t%s\t%s\n", db.Name, cq.Name, cq.Query))
```
Comment (danxmoran) [inline cmd/influxd/upgrade/database.go:195]: Other ideas I had for the export format:
1. Mimic the pseudo-TSV output of `SHOW CONTINUOUS QUERIES`, which looks like:
    ```
    > SHOW CONTINUOUS QUERIES
    name: _internal
    name query
    ---- -----

    name: mydb
    name     query
    ----     -----
    other_cq CREATE CONTINUOUS QUERY other_cq ON mydb BEGIN SELECT mean(foo) INTO mydb.autogen.bar FROM mydb.autogen.baz GROUP BY time(1h) END

    name: telegraf
    name     query
    ----     -----
    cq_basic CREATE CONTINUOUS QUERY cq_basic ON telegraf BEGIN SELECT mean(usage_user) INTO telegraf.autogen.average_user_cpu FROM telegraf.autogen.cpu GROUP BY time(1m) END
    ```
2. Output InfluxQL-compatible lines, with one CQ per line (and any extra metadata in comments):
    ```
    -- some metadata we feel like adding
    CREATE CONTINUOUS QUERY cq_basic ON telegraf BEGIN SELECT mean(usage_user) INTO telegraf.autogen.average_user_cpu FROM telegraf.autogen.cpu GROUP BY time(1m) END
    -- some other metadata
    CREATE CONTINUOUS QUERY other_cq ON mydb BEGIN SELECT mean(foo) INTO mydb.autogen.bar FROM mydb.autogen.baz GROUP BY time(1h) END
    ```

File: cmd/influxd/upgrade/database_test.go [hunk 2]
```diff
@@ -37,6 +37,7 @@ func TestUpgradeRealDB(t *testing.T) {
 
 	boltPath := filepath.Join(tl.Path, bolt.DefaultFilename)
 	enginePath := filepath.Join(tl.Path, "engine")
+	cqPath := filepath.Join(tl.Path, "cq.txt")
```
Comment (danxmoran) [inline cmd/influxd/upgrade/database_test.go:40]: Would it be worth adding some CQs to the checked-in test data here, so we can test against it?

File: cmd/influxd/upgrade/database.go (hunk 1 above)
Comment (stuartcarnie) [reply cmd/influxd/upgrade/database.go:195]: @timhallinflux do you have an preferences for the output format of the file? I'm inclined to write only the `CREATE CONTINUOUS` statements and nothing else.

Currently @danxmoran is writing a tab-delimited file as follows:

```tsv
database	name	query
telegraf	cq_basic	CREATE CONTINUOUS QUERY cq_basic ON telegraf BEGIN SELECT mean(usage_user) INTO telegraf.autogen.average_user_cpu FROM telegraf.autogen.cpu GROUP BY time(1m) END
```

File: cmd/influxd/upgrade/database_test.go (hunk 2 above)
Comment (stuartcarnie) [reply cmd/influxd/upgrade/database_test.go:40]: Having tests is the best way to ensure it isn't broken in the future 👍

**Review by stuartcarnie — COMMENTED** [review]
Awaiting @timhallinflux's comments on the format of the file. My preference is to write the queries only.

File: cmd/influxd/upgrade/database.go (hunk 1 above)
Comment (danxmoran) [reply cmd/influxd/upgrade/database.go:195]: I would also prefer writing just the `CREATE CONTINUOUS` statements

Comment (danxmoran) [reply cmd/influxd/upgrade/database.go:195]: Tim voted in Slack for mirroring the `SHOW CONTINUOUS QUERIES` output, moving ahead with that option

**Review by stuartcarnie — APPROVED** [review]
This is great 💯

**Pull request merged**